main.tf

```diff
@@ -133,6 +133,18 @@ resource "aws_route" "database_internet_gateway" {
   }
 }
 
+resource "aws_route" "database_nat_gateway" {
+  count = "${var.create_vpc && var.enable_nat_gateway && var.create_database_subnet_route_table && length(var.database_subnets) && var.create_database_nat_gateway_route && !var.create_database_internet_gateway_route > 0 ? local.nat_gateway_count : 0}"
```

Comment:
There is an error in expression. Should be `"${var.create_vpc && var.enable_nat_gateway && var.create_database_subnet_route_table && length(var.database_subnets) > 0 && var.create_database_nat_gateway_route && !var.create_database_internet_gateway_route ? local.nat_gateway_count : 0}"`.

The rest looks good.